### PR TITLE
PERF: op(frame, series) when series is not EA

### PR DESCRIPTION
**Pull request by jbrockmendel**
We have the same optimization in place for the flex ops.

ipython results for the asv this adds:

```
In [3]: arr = np.arange(10 ** 6).reshape(100, -1)                                                                                                                                                                   
In [4]: df = pd.DataFrame(arr)                                                                                                                                                                                         
In [5]: df["C"] = 1.0                                                                                                                                                                                               
In [6]: row = df.iloc[0]                                                                                                                                                                                            

In [11]: %timeit df + row                                                                                                                                                                                           
1.92 s ± 71.1 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)   # <-- master
1 ms ± 19.2 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)  # <-- PR
```

### CODE REVIEW

**Comment by jbrockmendel** [comment]
@WillAyd any guesses why the benchmark check would be exiting at 50%?

**Comment by jbrockmendel** [comment]
hmm when i run  `asv dev` locally it stops at 50% but with exit code 0.  any guesses @datapythonista ?

**Comment by datapythonista** [comment]
> hmm when i run `asv dev` locally it stops at 50% but with exit code 0. any guesses @datapythonista ?

I tried `asv run` locally with master, and it worked well until the end. Is it just `asv dev`, or just this PR that is stopping half way? I don't see why this should happen.

**Review by jreback — CHANGES_REQUESTED** [review]
lgtm. can you add a whatsnew note & an asv that covers this.

**Comment by datapythonista** [comment]
Tried `asv dev` locally, and also stopped at 50%. Is there an issue for it? I see that last release of `asv` was a year ago, so I guess it's something in our side.

**Comment by jbrockmendel** [comment]
Looks like the asv issue is actually that this causes an unrelated benchmark to fail.  Am troubleshooting now and will add an appropriate test.

**Comment by jbrockmendel** [comment]
> lgtm. can you add a whatsnew note & an asv that covers this.

done

**Review by jreback — APPROVED** [review]

File: pandas/core/ops/__init__.py [hunk 1]
```diff
@@ -522,6 +522,16 @@ def _combine_series_frame(left, right, func, axis: int, str_rep: str):
         new_data = dispatch_to_series(left, right, func)
 
     else:
+        rvalues = right._values
```
Comment (jreback) [inline pandas/core/ops/__init__.py:525]: can you put this in an elif (and leave out 533, i believe you can just set new_data=bm, then leave line 535 as the else

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/ops/__init__.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/ops/__init__.py:525]: ATM this is mirroring the code in the axis==0 case above; in a follow-up we'll de-duplicate these two chunks

**Review by jreback — APPROVED** [review]

**Pull request merged**